core/src/apps/cardano/layout.py

```diff
@@ -104,103 +252,148 @@ async def confirm_sending_token(
     )
 
 
-async def show_warning_tx_output_contains_tokens(ctx: wire.Context) -> None:
-    await confirm_metadata(
-        ctx,
-        "confirm_tokens",
-        title="Confirm transaction",
-        content="The following\ntransaction output\ncontains tokens.",
-        larger_vspace=True,
-        br_code=ButtonRequestType.Other,
+async def show_set_output_payment_credential(
+    ctx: wire.Context,
+    path: list[int],
+    script_hash: bytes | None,
+    address_type: CardanoAddressType,
+    credential_policy: int,
+) -> None:
+    await show_set_credential(
+        ctx, path, None, script_hash, None, address_type, credential_policy, "payment"
     )
 
 
-async def show_warning_path(ctx: wire.Context, path: list[int], title: str) -> None:
-    await confirm_path_warning(ctx, address_n_to_str(path), path_type=title)
-
-
-async def show_warning_tx_no_staking_info(
-    ctx: wire.Context, address_type: CardanoAddressType, amount: int
+async def show_set_output_stake_credential(
+    ctx: wire.Context,
+    path: list[int],
+    key_hash: bytes | None,
+    script_hash: bytes | None,
+    pointer: CardanoBlockchainPointerType | None,
+    address_type: CardanoAddressType,
+    credential_policy: int,
 ) -> None:
-    atype = ADDRESS_TYPE_NAMES[address_type].lower()
-    content = "Change %s address has no stake rights.\nChange amount:\n{}" % atype
-    await confirm_metadata(
+    await show_set_credential(
         ctx,
-        "warning_staking",
-        title="Confirm transaction",
-        content=content,
-        param=format_coin_amount(amount),
-        hide_continue=True,
-        br_code=ButtonRequestType.Other,
+        path,
+        key_hash,
+        script_hash,
+        pointer,
+        address_type,
+        credential_policy,
+        "stake",
     )
 
 
-async def show_warning_tx_pointer_address(
+async def show_set_credential(
```

Comment:
Since the function takes multiple parameters and only of them is "set" i.e. not None, I wanted to express that in the name of the function and I couldn't think of a better alternative. Maybe `show_credential` would suffice. A similar name is used also in `get_set_credential` and `get_set_credential_title`.

Reply:
I'd go with just `show_credential`, it's much less confusing.

Reply:
https://github.com/vacuumlabs/trezor-firmware/pull/66/commits/2dde1ce84799b9aac6a62bed084abac071a5a760#diff-718c35fab848e1ee6f5e2a387c71b0a0ea8d84039fedce530cb9f0584c164495R288